tests/lib/devicetree/api/app.overlay

```diff
@@ -344,6 +344,11 @@
 			misc-prop = <1234>;
 		};
 
+		test_vendor: vendor {
+			compatible = "vnd,model1", "vnd,model2", "gpio";
```

Comment:
I think it might be good to re-order the list as:

`compatible = "vnd,model1", "gpio", "vnd,model2";`

That way the index test might not miss an off-by-one

Reply:
Probably good to use some other non-`vnd` in this list as well.  maybe have a `zephyr,model2` or something.